estimagic/optimization/optimize.py

```diff
@@ -318,13 +417,36 @@ def _internal_minimize(
             Set to -1 for maximizations to plot the fitness that is being maximized.
 
     """
+    logging_decorator = functools.partial(
```

Comment:
In my next PR I will replace the `queue` argument of `create_internal_criterion` by the `database`. Thus I suggest you already add `database` as argument now. Then we don't need `logging_decorator` as argument because we have all available information to construct it inside `create_internal_criterion`.

Reply:
I have implemented ``logging_decorator`` because the ``internal_criterion`` for the function evaluations receives a different decorator than the ``internal_criterion`` for the gradient.

Reply:
I understand. That's a good solution.